### PR TITLE
Implement simple dbus service

### PR DESCRIPTION
**Pull request by miltonio94**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,89 +1,100 @@
...
-        conn.process(Duration::from_millis(1000)).unwrap();
+    // async fn go_away(&self,
+    // #[zbus(signal_context)]
+    // ctxt: SignalContext<'_>,
+    // ) -> fdo::Result<()> {
+    //     Self::greeted_everyone(&ctxt).await?;
+    //     self.done.notify(1);
+
+    //     Ok(())
+    // }
```
Comment (murpjack) [inline src/main.rs:65]: Think these commented lines can be removed

File: src/main.rs [hunk 2]
```diff
@@ -1,89 +1,100 @@
...
-                    eprintln!("Failed to eavesdrop: '{}', trying without it", e);
-                    conn.add_match(rule, |_: (), _, msg| {
-                        handle_message(&msg);
-                        true
-                    })
-                    .expect("add_match failed");
-                }
-            }
-        }
+use zbus::{Connection, Result, dbus_interface, ConnectionBuilder, SignalContext, fdo};
```
Comment (murpjack) [inline src/main.rs:1]: Awesome!

File: src/main.rs [hunk 3]
```diff
@@ -1,89 +1,100 @@
...
-    println!("Got message: {:?}", msg);
+#[tokio::main]
+async fn main() -> Result<()> {
+    let state: State = State {
+        transactions: Vec::new()
+    };
+
+
+    let _connection = ConnectionBuilder::session()?
+        .name("org.green_sharkd.GreenSharkd")?
```
Comment (murpjack) [inline src/main.rs:89]: This area of the code makes a great deal more sense to me now

File: src/main.rs [hunk 4]
```diff
@@ -1,89 +1,100 @@
...
+
+#[derive(Debug)]
+struct State {
+    transactions: Vec<Transaction>
+}
+
+#[dbus_interface(name = "org.green_sharkd.Commands")]
+impl State {
+    async fn add_transaction(&mut self, amount: f32, name: &str) -> String {
+        println!("adding transaction");
```
Comment (murpjack) [inline src/main.rs:40]: I like that we are informing the end user while adding

File: src/main.rs (hunk 1 above)
Comment (miltonio94) [reply src/main.rs:65]: Thanks @murpjack I was thinking we could leave it in there for now for us to reference it but I think you are right, we can get rid of them. We have enough to go forward

**Review by murpjack — CHANGES_REQUESTED** [review]
Looks good. Please remove commented lines

File: src/main.rs (hunk 1 above)
Comment (miltonio94) [reply src/main.rs]: I've done that now

**Review by murpjack — APPROVED** [review]

**Pull request merged**